### PR TITLE
[Instrumentation.EntityFrameworkCore] Stop emitting db.statement_type attribute

### PR DESCRIPTION
**Pull request by Kielek**
## Changes

Similar changes to https://github.com/open-telemetry/opentelemetry-dotnet/pull/5301
[Instrumentation.EntityFrameworkCore] Stop emitting `db.statement_type` attribute

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.EntityFrameworkCore/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Stop emitting `db.statement_type` attribute.
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.EntityFrameworkCore/CHANGELOG.md:5]:
```suggestion
* **Breaking Change** : Stop emitting `db.statement_type` attribute.
```

**Review by vishweshbankwar — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.EntityFrameworkCore/CHANGELOG.md (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.EntityFrameworkCore/CHANGELOG.md]: 330172166f426af017619b64736851883fbbfc8b

**Comment by julealgon** [comment]
Would be nice if you guys could add a bit more detail on the release notes as to _why_ a change is being made, especially breaking ones.

This is referred in the release notes with just:
> Breaking Change: Stop emitting db.statement_type attribute

Even this PR has zero information, and the linked one for the SQL instrumentation is fairly barebones.

**Comment by cijothomas** [comment]
> Would be nice if you guys could add a bit more detail on the release notes as to _why_ a change is being made, especially breaking ones.
> 
> This is referred in the release notes with just:
> 
> > Breaking Change: Stop emitting db.statement_type attribute
> 
> Even this PR has zero information, and the linked one for the SQL instrumentation is fairly barebones.

That is fair feedback.
The reason is - that attribute added at a time when conventions were being written, and the attribute never got part of any conventions. So the change is done to align with the conventions.

**Pull request merged**